users/engler/fairseq/training.py

```diff
@@ -0,0 +1,162 @@
+import os
+import subprocess as sp
+import yaml
+import sys
+
+from sisyphus import *
+
+import recipe.i6_core.util as util
+
+
+class FairseqHydraConfig:
+    """
+    An object that manages a Fairseq hydra config (inspired by the ReturnnConfig).
+    """
+
+    def __init__(self, config_dict, *, yaml_prefix=""):
+        """
+        :param dict config_dict: Contains the information which is needed for fairseq-hydra-train. Will be converted and dumped into a .yaml
+        :param str yaml_prefix: Prefix which should be written to the beginning of the config, for example "# @package _group_"
+        """
+        assert isinstance(config_dict, dict)
+        self.config_dict = config_dict
+        self.yaml_prefix = yaml_prefix
+
+    def write(self, path):
+        config_yaml = yaml.dump(self.config_dict)
+        # "# @package _group_" was written at the beginning in the example .yaml from fairseq:
+        if self.yaml_prefix != "":
+            config_yaml = self.yaml_prefix + "\n" + config_yaml
+        with open(path, "w") as file:
+            file.write(config_yaml)
+
+
+class PytorchHydraModel:
+    """
+    Defines a Pytorch hydra model as yaml config, pytorch checkpoint file and epoch
+    """
+
+    def __init__(self, fairseq_hydra_config_file, model, epoch):
+        """
+
+        :param Path fairseq_hydra_config_file: Path to a returnn config file
+        :param Path model: Path to a pytorch checkpoint
+        :param int epoch:
+        """
+        self.returnn_config_file = fairseq_hydra_config_file
+        self.model = model
+        self.epoch = epoch
+
+
+class FairseqHydraTrainingJob(Job):
+    """
+    Train a Fairseq model using fairseq-hydra-train
+    """
+
+    def __init__(
+        self,
+        fairseq_hydra_config,
+        *,  # args below are keyword only
+        command_line_args=None,
+        max_epoch=1,
+        save_interval=1,
+        keep_epochs=None,
+        time_rqmt=4,
+        mem_rqmt=4,
+        cpu_rqmt=2,
+        gpu_rqmt=1,
+        fairseq_python_exe=None,
+        fairseq_hydra_exe=None,
+        fairseq_git=None,
```

Comment:
the consistent name to the other jobs would be `fairseq_root`, because in theory the source folder can come from non-git sources.